lib/agent.js

```diff
@@ -144,37 +144,41 @@ Agent.prototype.start = function (opts) {
     })
   }
 
-  this._apmServer = new ElasticAPMHttpClient({
-    // metadata
-    agentName: 'nodejs',
-    agentVersion: version,
-    serviceName: this._conf.serviceName,
-    serviceVersion: this._conf.serviceVersion,
-    frameworkName: this._conf.frameworkName,
-    frameworkVersion: this._conf.frameworkVersion,
-    hostname: this._conf.hostname,
-
-    // Sanitize conf
-    truncateKeywordsAt: config.INTAKE_STRING_MAX_SIZE,
-    truncateErrorMessagesAt: this._conf.errorMessageMaxLength,
-
-    // HTTP conf
-    secretToken: this._conf.secretToken,
-    userAgent: userAgent,
-    serverUrl: this._conf.serverUrl,
-    rejectUnauthorized: this._conf.verifyServerCert,
-    serverTimeout: this._conf.serverTimeout * 1000,
-
-    // Streaming conf
-    size: this._conf.apiRequestSize,
-    time: this._conf.apiRequestTime * 1000,
-
-    // Debugging
-    payloadLogFile: this._conf.payloadLogFile
-  })
-  this._apmServer.on('error', err => {
-    this.logger.error('An error occrued while communicating with the APM Server:', err.message)
-  })
+  if (this._conf.transport) {
+    this._apmServer = this._conf.transport(this._conf)
```

Comment:
Thinking if we should rename `_apmServer` to `_transport` 🤔

Reply:
Renamed to `_transport`.